test/test_multispecies_problem.py

```diff
@@ -0,0 +1,139 @@
+import numpy as np
+import festim as F
+from petsc4py import PETSc
+from dolfinx.fem import Constant
+from ufl import exp
+import os
+
+
+def test_multispecies_permeation_problem():
+    """Test running a problem with 2 mobile species permeating through a 1D
+    0.3mm domain, with different diffusion coefficients, asserting that the
+    resulting concentration fields are less than 1% different from their
+    respecitive analytical solutions"""
+
+    # festim model
+    L = 3e-04
+    my_mesh = F.Mesh1D(np.linspace(0, L, num=1001))
+    my_model = F.HydrogenTransportProblem()
+    my_model.mesh = my_mesh
+
+    my_mat = F.Material(
+        D_0={"spe_1": 1.9e-7, "spe_2": 3.8e-7},
+        E_D={"spe_1": 0.2, "spe_2": 0.2},
+        name="my_mat",
+    )
+    my_subdomain = F.VolumeSubdomain1D(id=1, borders=[0, L], material=my_mat)
+    left_surface = F.SurfaceSubdomain1D(id=1, x=0)
+    right_surface = F.SurfaceSubdomain1D(id=2, x=L)
+    my_model.subdomains = [
+        my_subdomain,
+        left_surface,
+        right_surface,
+    ]
+
+    spe_1 = F.Species("spe_1")
+    spe_2 = F.Species("spe_2")
+    my_model.species = [spe_1, spe_2]
+
+    temperature = Constant(my_mesh.mesh, 500.0)
+    my_model.temperature = temperature
+
+    my_model.boundary_conditions = [
+        F.DirichletBC(subdomain=right_surface, value=0, species="spe_1"),
+        F.DirichletBC(subdomain=right_surface, value=0, species="spe_2"),
+        F.SievertsBC(
+            subdomain=left_surface, S_0=4.02e21, E_S=1.04, pressure=100, species="spe_1"
+        ),
+        F.SievertsBC(
+            subdomain=left_surface, S_0=4.02e21, E_S=1.04, pressure=100, species="spe_2"
```

Comment:
Could you use 2 different solubilities to make sure we are correctly applying the bcs?